src/main/java/gregtech/common/covers/CoverItemVoidingAdvanced.java

```diff
@@ -0,0 +1,164 @@
+package gregtech.common.covers;
+
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.cover.ICoverable;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.Widget;
+import gregtech.api.gui.widgets.*;
+import gregtech.client.renderer.texture.Textures;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class CoverItemVoidingAdvanced extends CoverItemVoiding {
+
+    protected VoidingMode voidingMode;
+
+    public CoverItemVoidingAdvanced(ICoverable coverHolder, EnumFacing attachedSide) {
+        super(coverHolder, attachedSide);
+        this.voidingMode = VoidingMode.VOID_ANY;
+        this.itemFilterContainer.setMaxStackSize(1);
+    }
+
+    @Override
+    protected void doTransferItems() {
+        IItemHandler myItemHandler = coverHolder.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, attachedSide);
+        if (myItemHandler == null) {
+            return;
+        }
+
+        switch (voidingMode) {
+            case VOID_ANY:
+                voidAny(myItemHandler);
+            case VOID_OVERFLOW:
+                voidExact(myItemHandler);
+        }
+    }
+
+    void voidAny(IItemHandler myItemHandler) {
+        for (int srcIndex = 0; srcIndex < myItemHandler.getSlots(); srcIndex++) {
+            ItemStack sourceStack = myItemHandler.extractItem(srcIndex, Integer.MAX_VALUE, true);
+            if (sourceStack.isEmpty()) {
+                continue;
+            }
+            if (!itemFilterContainer.testItemStack(sourceStack)) {
+                continue;
+            }
+            myItemHandler.extractItem(srcIndex, Integer.MAX_VALUE, false);
+        }
+    }
+
```

Comment:
Is this needed? It looks to be the exact same as `CoverItemVoiding#voidAny`, which this filter has access to